engine/crates/engine-v2/src/sources/graphql/mod.rs

```diff
@@ -5,5 +5,45 @@ mod request;
 mod root_fields;
 mod subscription;
 
+use std::time::Duration;
+
 pub(crate) use federation::*;
+use headers::HeaderMapExt;
 pub(crate) use root_fields::*;
+
+use grafbase_telemetry::gql_response_status::GraphqlResponseStatus;
+use runtime::bytes::OwnedOrSharedBytes;
+
+fn should_update_cache(status: GraphqlResponseStatus, cache_control: Option<&headers::CacheControl>) -> bool {
+    status.is_success()
+        && cache_control
```

Comment:
I would have moved that to the TTL computation, if `no-store` / `private`, would return a `None` for the TTL. Would keep the `Cache-Control` header manipulation in a single place. And would have provided only the headers to the `calculate_cache_ttl`, looked a bit weird to provide the full http response rather than just the headers IMHO.